Ccd windows philippines

gavram opened the pull request:
@zelima 

I've finished this task as you requested. What I did?
1. I created two new files:
  * `.docker-compose.vital-strategies-philippines-theme.yaml`
  * `/docker-compose/ckan-conf-templates/vital-strategies-philippines-theme-production.ini.template`
2. Modifications are made in two more files
  * `/ckan/Dockerfile`
  * `docker-compose.yaml`
3. `DEPLOYing.md` is refactored

The resulting CKAN instance with Vital Strategies theme in browser on host Windows Server 2008 looks as on following screenshot

![image](https://user-images.githubusercontent.com/12686547/83640858-0630ad00-a5ad-11ea-8cfa-02c58192628b.png)

The instance is tested:
1. creating new org
2. creating new goup
3. creating new dataset
4. uploading new resource
5. updating resource
and all is working well.

gavram committed:
[modification][s] CCD Vital Strategies Philippines installation on Windows Server 2008

gavram committed:
[philippines] Deployment guide

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+* At least 30 GB of hardrive free space for initial installation of VirtualBox, Ubuntu 18.04 inside VM and CCD with Vital Strategies template.
 
-First we need to install **[git](https://help.github.com/en/articles/set-up-git)** and **[docker-compose](https://docs.docker.com/compose/install/)** \(*docker-compose* should already have *docker* as dependency. If this is not the case follow the **[official documentation on installing docker](https://docs.docker.com/v17.12/install/)**\):
+---
+---
```
Think we don't need this

gavram replied:
I'll remove that lines

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+---
+---
+
+Installation of CCD Vital Strategies Philippines on Ubuntu 18.04 LTS inside Virtual Box VM
```
Maybe 
> Installation guide for CKAN instance for Philipines project on ...

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+3. Ubuntu 18.04 LTS inside VM installation
+4. Additional settings for Ubuntu in VM
+5. CCD Vital Strategies installation
+6. Exposing CKAN Vital Strategies to outside World
```
Let's either make this table of content or remove. They can simply read as they go along.  No need to list in the begging unless they are clickable that will take you to specific place.

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+5. CCD Vital Strategies installation
+6. Exposing CKAN Vital Strategies to outside World
+
+## 1. VirtualBox installation
```
Make this 3rd level heading 

`###`

Also no need of 1.

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+## 1. VirtualBox installation
+
+#### 1 - Download VirtualBox installer for Windows
```
Remove numbers

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+Go to the page above and download the binary version for Windows hosts
+
+#### 2 - Run the installer and follow the instructions
```
Remove number

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+![](https://i.imgur.com/47Qk8zV.png)
+
+#### 3 – Custom setup dialog box
```
Remove number

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+#### 3 – Custom setup dialog box
+
+You will see custom setup dialog box. There is not much to choose from. You can accept the default and click next.
+If you wish to change the installation directory, you van change it by clicking on the browse button and selecting the new directory and clicking OK. Normally I leave it as the default as the whole installation process does not take much space on your hard drive.
```
> van -?

Let's not leave them the choice but give exact directives... Smth like, just leave everything to defaults..

zelima replied:
@gavram How about this?

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+![](https://i.imgur.com/eJfBrNG.png)
+
+VirtualBox Installation – Custom setup dialog box screenshot
```
What is this about?

zelima replied:
Also this

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+VirtualBox Installation – Custom setup dialog box screenshot
+
+#### 4 – Custom setup dialog box – Feature to install
```
Remove number

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+#### 4 – Custom setup dialog box – Feature to install
+
+In this dialog box you can choose which features to install. As you can see, there is not much to choose. You can accept the default and click next. You can uncheck Create a shortcut in the quick launch bar and register file association.
```
Just say accept default

zelima replied:
^

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+![](https://i.imgur.com/R7aIe7b.png)
+
+VirtualBox Installation – Custom Setup – Select feature to Install
```
Is this title (and one above)? If so make it 3 level heading

zelima replied:
^

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+VirtualBox Installation – Custom Setup – Select feature to Install
+
+#### 5 – Network Interface setup
```
Remove number

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+![](https://i.imgur.com/NhpgPPG.png)
+
+VirtualBox Installation – Network Interface warning
```
The same here make it 3rd level heading if is title

zelima replied:
^

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+VirtualBox Installation – Network Interface warning
+
+#### 6 – Ready to Install
```
same

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+![](https://i.imgur.com/WGvTus0.png)
+
+VirtualBox Installation – Ready to Install
```
same

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+VirtualBox Installation – Ready to Install
+
+#### 7 – Installation begins
```
Same

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+VirtualBox Installation in Progress
+
+#### 8 – Installation Completes
```
same

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+#### 8 – Installation Completes
+
+After the installation completes, you will see installation completion dialog box. Click finish. If you leave Start Oracle VM VirtualBox after installation checked, VirtualBox will launch automatically. If you uncheck it, you will have to launch Virtualbox manually. Its not big deal anyways. Click finish to exit installation.
```
same here don't leave them a choice. Just say what to do

zelima replied:
^

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+VirtualBox Installation complete
+
+---
+---
```
No need of these

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+Your VM for Ubuntu installation is created and setup.
+
+---
+---
```
remove these

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+* Click on `Restart Now`
+
+---
+---
```
remove

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+---
+---
+
+## 4. Additional settings for Ubuntu in VM
```
remove number

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+## 4. Additional settings for Ubuntu in VM
+
+#### 1. Git installation
```
make it 3rd level heading

zelima replied:
and remove number

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
 
-Then start and enable the docker service and verify operation
 
+#### 2. `docker-compose` installation
```
remove the number and make it 3rd level

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+---
+---
+
+## 5. CCD Vital Strategies installation
```
remove number

Say CKAN installation (they don't have clue what is CCD vital strategies :) )

zelima commented on DEPLOYING.md:
````diff
@@ -1,62 +1,351 @@
-#### Extra dependencies
+3. Change branch to `ccd-windows-philippines`
+```
+git checkout ccd-windows-philippines
````
Think they can directly clone the branch, Smth like

```
git clone -b <<branch-name>> url
```

zelima commented on DEPLOYING.md:
````diff
@@ -1,62 +1,351 @@
+```
 
-- You will also need a SMTP server and its credentials for CKAN to work properly. This will not obstacle deployment, CKAN will be up and running, but won't be able to send emails (e.g. on password reset). You will be asked to provide SMTP server credentials while running `make secret` script, see below.
+4. Run
````
Say why run. Eg `To create secrets run`

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+Hit `Enter` on every question.
 
-Now we have the runtime, next we need to download the cluster configuration files and build the services.
+5. Run
```
say `Build the image and start the application`

zelima replied:
^

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+and wait to finish.
+
+7. In the folowing files change `CKAN_BRANCH:-ckan-2.7.3` to `CKAN_BRANCH:-ckan-2.8.1`:
+  * **`docker-compose.yaml`** - line #103
+  * **`.docker-compose.vital-strategies-philippines-theme.yaml`** - lines #24 and #43
+  * **`ckan/Dockerfile`** - line #61
```
I'm still concerned about this part. Let's either make it 2.8.X from the begging or debug. What happens if you have 2.8.X all the time?

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+![](https://i.imgur.com/PSizKmS.png)
+
+---
+---
```
remove this

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+---
+---
+
+## 6. Exposing CKAN Vital Strategies to outside World
```
Remove number and say "Expose CKAN to the outside world."

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+Here we should noticed that ip address of our Ubuntu VM in this case is `192.168.56.1`.
+**Now we can access our CKAN server from host Windows Server 2008 by visiting `192.168.56.1:8080`**
+
+### 2. Using `bridge adapter` method
```
Let's pick one of them... This one's suggested by Adam as much as I remember so let's keep this

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+Now you can access your CKAN instance from host Windows Server 2008 by visiting `192.168.1.103:8080`
+
+---
+---
```
remove

zelima commented on DEPLOYING.md:
```diff
@@ -1,62 +1,351 @@
+
+For that, we only need to create `inbound` rule in Windows Firewall for port `8080` to allow access through that port.
+
+---
+---
+---
```
remove

zelima commented on ckan/Dockerfile:
```diff
@@ -58,7 +58,7 @@ USER ckan
 ARG CKAN_REPO
 
-RUN CKAN_BRANCH="${CKAN_BRANCH:-ckan-2.8.1}" && CKAN_REPO="${CKAN_REPO:-ckan/ckan}" &&\
+RUN CKAN_BRANCH="${CKAN_BRANCH:-ckan-2.7.3}" && CKAN_REPO="${CKAN_REPO:-ckan/ckan}" &&\
```
Let's have 2.8.1 (at least here). It will be overridden if ser in .docker-compos.philiphins..

zelima commented on docker-compose.yaml:
```diff
@@ -98,7 +100,7 @@ services:
       context: ckan
       args:
-        CKAN_BRANCH: ${CKAN_BRANCH:-ckan-2.8.1}
+        CKAN_BRANCH: ${CKAN_BRANCH:-ckan-2.7.3}
```
same here, worst case set it in .docker-compose.override...

gavram committed:
[refactoring][s] CCD Vital Strategies Philippines setup on Windows Server 2008

gavram commented:
@zelima I refactored the installation procedure. I made the changes you requested. Everything is tested several times and all is working well - better than in previous procedure.
Also, I've corrected formatting in `DEPLOYING.md` file and made some simplifications in the procedure.

zelima commented:
@gavram missed a few comments. marked with ^

gavram committed:
[fix][s] Content fixes

gavram commented:
@zelima I've fixed content of `DEPLOYING.md` file that I missed in previous commit.
